xray/client/xray_viewer.ts

```diff
@@ -183,6 +195,9 @@ export class XRayViewer {
   }
 
   private swapIn(nodeId: string) {
+    if (this.cleanupCalled) {
```

Comment:
What is this actually guarding against? It looks like the texture loading callback is already guarded by `inScene`, so it seems like it's only `nodesUpdate` that could call into here. If so, I think it makes more sense to move this check to there.

Reply:
This guards that no new node is swapped in, which might happen when an async load returns. Once we cleaned up and removed all nodes from the scene, we defintively do not want to add more in, so I think this is the correct place to guard against - because here is where we swap in the new node.